loadbalancer/algorithm.go

```diff
@@ -89,19 +95,71 @@ func (*consistentHash) Apply(ctx *routing.LBContext) routing.LBEndpoint {
 	key := net.RemoteHost(ctx.Request).String()
 	if _, err := h.Write([]byte(key)); err != nil {
 		log.Errorf("Failed to write '%s' into hash: %v", key, err)
-		return ctx.Route.LBEndpoints[rand.Intn(len(ctx.Route.LBEndpoints))]
+		return ctx.Route.LBEndpoints.At(rand.Intn(ctx.Route.LBEndpoints.Length()))
 	}
 	sum = h.Sum32()
-	choice := int(sum) % len(ctx.Route.LBEndpoints)
+	choice := int(sum) % ctx.Route.LBEndpoints.Length()
 	if choice < 0 {
-		choice = len(ctx.Route.LBEndpoints) + choice
+		choice = ctx.Route.LBEndpoints.Length() + choice
+	}
+	return ctx.Route.LBEndpoints.At(choice)
+}
+
+type powerOfChoices struct {
```

Comment:
I think "power" in the  "The Power of Two Choices in Randomized Load Balancing" implies strength of two or N random choices over single one.

I personally would go with something casual like `randomN` or in this particular case even with `random2` as number of choices is hardcoded.

Reply:
Note that I am a bystander here, not a codeowner so consider this a nitpick :)

Reply:
I was wondering to change it to `powerOfNChoices` or `randomNChoices`. I prepare the code for experimenting with the number of choices. I would like to keep the word `choices` as it is a popular word reference for the algorithm.

Reply:
I renamed it to PowerOfRandomNChoices. It is better to be explicit. If people want to research it is a googleable.

Reply:
> If people want to research it is a googleable.

Or just add a link to the paper in the comment :)

Reply:
I will also do that.